src/main/java/uk/nhs/digital/nhsconnect/nhais/translator/FhirToEdifactManager.java

```diff
@@ -31,15 +36,15 @@
     }
 
     private List<Segment> delegateAcceptance(Parameters parameters) throws FhirValidationException {
-        String acceptanceType = determineAcceptanceType(parameters);
-        if("birth".equals(acceptanceType)) {
-            return acceptanceBirthTranslator.translate(parameters);
+        var acceptanceType = ParametersExtension.extractAcceptanceType(parameters);
```

Comment:
TODO from 'createMessageSegments' might be removed now?

Reply:
no, it's needed